### PR TITLE
refactor NB for softcoded sfreq

### LINKED ISSUE #37: Unconsistent number of epochs after cropping of mne.raw
I think you would guess that these lines give the same output:

1) 
```
raw  = mne.io.read_raw_edf(filename, preload=False)  # prepare loading
raw  = CreateRaw(raw[picks_chan][0], picks_chan, ch_types=['emg'])        # pick channels and load
raw  = raw.load_data()  # load data into memory 

croptimes=dict(tmin=raw.times[0]+3600*2, tmax=raw.times[-1]-3600)
raw.crop(**croptimes)

sfreq = raw.info["sfreq"]
window_length = 0.25                    # in seconds
duration = int(window_length * sfreq)   # in samples
interval = duration                     # no overlapping
epochs = RawToEpochs_sliding(raw, duration=duration, interval=interval)
print(f"Epochs done, shape {epochs.shape}")
```
REPONSE: Epochs done, shape (115700, 2, 50)

2) 
```
raw  = mne.io.read_raw_edf(filename, preload=False)  # prepare loading

croptimes=dict(tmin=raw.times[0]+3600*2, tmax=raw.times[-1]-3600)
raw.crop(**croptimes)

raw  = CreateRaw(raw[picks_chan][0], picks_chan, ch_types=['emg'])        # pick channels and load
raw  = raw.load_data()  # load data into memory 



sfreq = raw.info["sfreq"]
window_length = 0.25                    # in seconds
duration = int(window_length * sfreq)   # in samples
interval = duration                     # no overlapping
epochs = RawToEpochs_sliding(raw, duration=duration, interval=interval)
print(f"Epochs done, shape {epochs.shape}")
```
REPONSE: Epochs done, shape (104900, 2, 50)

Odd isn't? If you know why, don't hesitate to tell me!

### LINKED ISSUE #84: Dealing with unconsistent sfreq
It seems that after refactoring all from 200 to 250Hz (#61 ), the problem was that half of the recording are 200Hz and the other half to 250Hz.

We can either:
### OPTION 1 not dealing with different sample rate (e.g. changing duration)**
pro:
- more scalable

con:
- number of samples per epochs will be different which can modify a bit the results

### OPTION 2 re-sampling all to 200 Hz**
pro:
- all epochs for all subjects are consistent

con:
- computationally intensive
- non-proportional resampling interpolation could actually modify the statistical properties of the signals.

Anyway, there is no-clear best practice here but I actually prefer not to resample. Less struggle for OPTION 1.

### LINKED ISSUE #87: Small mistake in docstring
While making OMA verification, I used thee new function "episodes_to_list" from events.scoring.
I noticed that the argument n_labels is not properly described in docstring as it is written "n_labels : int
        duration of the recording in seconds" but it is not the duration of the recording in seconds (I tried, doesn't work), but rather the length of the classification labels list the list_episodes comes from.
It should be corrected in a future PR.

### LINKED ISSUE #88: Use Inter-Subject NB only for visualization
Preprocessing / Reporting could be embedded in a script with both Bruxism and MEMA analysis as both of common operations that are computed redundantly.

Similar as the NB combine, we could read the .pk file from the results.

Not a lot of work.

### LINKED ISSUE #89: Sometimes no possible estimation for power ratio estimation
Here is the error I get while trying to run the bruxism inter subject script on 1DA15_nuit son: 
```
---------------------------------------------------------------------------
ValueError                                Traceback (most recent call last)
~\Documents\acou_sommeil_HD_ENS\Tinnitus-n-Sleep\compute_results.py in <module>
    209                                                       time_interval=window_length_bruxism,
    210                                                       delim=delim, n_adaptive=n_adaptive_bruxism, log=log,
--> 211                                                       generate_report=generate_bruxism_report)
    212                     print(f"done)", end=" ")
    213                 else:

~\Documents\acou_sommeil_HD_ENS\Tinnitus-n-Sleep\tinnsleep\reports.py in reporting(epochs, valid_labels, THR_classif, time_interval, delim, n_adaptive, log, generate_report)
    258 
    259         report = generate_report(labels, time_interval, delim)
--> 260         report["Power Ratio"] = power_ratio(epochs[valid_labels], labels)
    261 
    262         labels = fuse_with_classif_result(np.invert(valid_labels),

~\Documents\acou_sommeil_HD_ENS\Tinnitus-n-Sleep\tinnsleep\signal.py in power_ratio(epochs, labels, axis)
     64     nb_epochs = len(labels)
     65     if (nb_num == 0) or (nb_num == nb_epochs):
---> 66         raise ValueError("labels should have at least one True and one False")
     67 
     68     pow = mean_power(epochs, axis=axis)

ValueError: labels should have at least one True and one False
```

This is not so surprising : we should expect that sometimes no events are spotted (although it is weird). It should work anyway. I'm going to put a condition for the calculus of the power ratio : return "nan" if there is always true or false.

### LINKED ISSUE #90: Artifact Detection uses too strict criteria
### observation
too much brux artefacts makes bruxism artifact very badly detected (see figure below)

log report show that `is_good` is the reponsible, here more than 6% of epochs are removed for HZB_nuit_2: 
```
                          bruxism           OMA       IMP
suppressed_is_good    6627.000000      0.000000       0.0
suppressed_amp_thr       0.000000    751.000000       0.0
suppressed_overall    6627.000000    751.000000       0.0
total_nb_epochs     104500.000000  26125.000000  104500.0
suppressed_ratio         0.063416      0.028746       0.0
```

### solution
more permissive threshold (e.g. `emg=1e-04` or `emg=5e-04` instead of `emg=5e-05`)

read new commits on PR #85  for changes and rerun script `compute_results`

```
![Screenshot 2020-05-27 at 16 41 56](https://user-images.githubusercontent.com/15196605/83034867-0541cd80-a039-11ea-9de8-078624a2fef1.png)

### PR DESCRIPTION
**Pull request by lkorczowski**
- [x] softcoded sfreq Bruxism inter
- [x] softcoded sfreq MEMA inter
- [x] refactor bruxism intra-subject
- [x] refactor MEMA intra-subject
- [x] clean combine NB
- [x] combine MEMA and bruxism computation into a single script #88 .
- [x] combine NB: fix management of bruxism and MEMA files when only one is available
- [x] `check_annotation.py` script for loading results and visualization of the annotations
- [x] merged annotation with new parameters `merge` in data.AnnotateRaw_sliding
- [x] remove power ratio error when to valid epoch #89 
- [x] updated script `compute_results.py` to deal with too strict artifact detection #90 

closes #37 
closes #84 
closes #87 
closes #88 
closes #89 
closes #90

### CODE REVIEW

**Comment by lkorczowski** [comment]
@RobinGuillard I couldn't make the analysis of the results for combine because of time but the rest is working. You need to run the script on your compute and push the new .pk first.

You can run the following script in the second cell of the combine NB:
```
%run ../compute_results.py --bruxism True --mema True --overwrite True 
```
Note the ` --overwrite True ` that will overwrite the results, without doing that, the script will do nothing.

File: compute_results.py [hunk 1]
```diff
@@ -0,0 +1,219 @@
...
+    results_file_bruxism = "data/reports_and_datas_bruxism.pk"
+
+
+    # Dictionnary of known names of the Airflow
+    mapping = {"Airflow": "MEMA"}
+    print("parameters set")
+
+    # Importing personnalized parameters for dataset
+    mema_files = pd.read_csv("data/mema_files.csv", engine='python', sep="; ")["files_with_mema"].values
+    dico_chans = pd.read_pickle("data/valid_chans_THR_imp.pk").to_dict("list")
```
Comment (RobinGuillard) [inline compute_results.py:68]: We should maybe verify one last time that the channel selection is accurately done

File: compute_results.py [hunk 2]
```diff
@@ -0,0 +1,219 @@
...
+                    print(f"done)", end=" ")
+                else:
+                    print(f"skipped)", end=" ")
+
+                print(f"MEMA(", end="")
+                if DO_MEMA:
+                    epochs_MEMA, valid_labels, log["MEMA"] = preprocess(raw, duration_MEMA, duration_MEMA,
+                                                                             picks_chan=['Airflow'])
+                    valid_labels_MEMA.append(valid_labels)
+                    print(f"done)", end=" ")
```
Comment (RobinGuillard) [inline compute_results.py:145]: What is the purpose of these lines? there shouldn't be a is_good processing on MEMA no?

File: notebooks/Bruxism_and_MEMA_combine.ipynb [hunk 3]
```diff
@@ -10,18 +10,14 @@
   },
```
Comment (RobinGuillard) [inline notebooks/Bruxism_and_MEMA_combine.ipynb:10]: <p>I remember now that we should also check why there were cases with combine bruxism without associated combine mema episode. it shouldn't exist!</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/lkorczowski/Tinnitus-n-Sleep/pull/85/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='5'/>

File: notebooks/MEMA_episode_visualisation.ipynb [hunk 4]
```diff
@@ -10,14 +10,18 @@
   },
```
Comment (RobinGuillard) [inline notebooks/MEMA_episode_visualisation.ipynb:10]: <p>Shouldn't we suppress this notebook ? It does not have any purpose nowadays and is no longer functionnal</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/lkorczowski/Tinnitus-n-Sleep/pull/85/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='1'/>

File: notebooks/Middle_ear_Inter_subject_analysis.ipynb [hunk 5]
```diff
@@ -43,45 +43,37 @@
     "import scipy\n",
```
Comment (RobinGuillard) [inline notebooks/Middle_ear_Inter_subject_analysis.ipynb]: <p>Unecessary definition of parameters : EDF_files, the name of result file hould be put in the cell where it is used for more covenience</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/lkorczowski/Tinnitus-n-Sleep/pull/85/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='2'/>

Comment (RobinGuillard) [inline notebooks/Middle_ear_Inter_subject_analysis.ipynb]: <p>This is not the result for number of episodes per hour but rather the power ratio, it should be recomputed</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/lkorczowski/Tinnitus-n-Sleep/pull/85/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='19'/>

File: tinnsleep/reports.py [hunk 6]
```diff
@@ -195,8 +195,12 @@ def merge_fun(foo):
...
     valid_labels = merge_fun(np.c_[np.invert(RMSlabels), amplitude_labels])
     suppressed_all = np.sum(np.invert(valid_labels))
-    log = {"suppressed_is_good": suppressed_is_good, "suppressed_amp_thr": suppressed_amp_thr,
-                                     "suppressed_overall": suppressed_all, "total_nb_epochs": len(valid_labels)}
+    log = {"suppressed_is_good": suppressed_is_good,
+           "suppressed_amp_thr": suppressed_amp_thr,
+           "suppressed_overall": suppressed_all,
+           "total_nb_epochs": len(valid_labels),
+           "suppressed_ratio": suppressed_all/len(valid_labels)}
+
```
[the remark this answers is not in the document]
Comment (lkorczowski) [reply tinnsleep/reports.py]: it is done.
to do so in NB:
`results[file]["log"]["OMA"][key_log]`

File: compute_results.py (hunk 2 above)
Comment (lkorczowski) [reply compute_results.py:145]: preprocess is just used here to pick the channel and do the epoching
valid_labels should be all True

File: compute_results.py (hunk 1 above)
Comment (lkorczowski) [reply compute_results.py:68]: added TODO

File: notebooks/MEMA_episode_visualisation.ipynb (hunk 4 above)
Comment (lkorczowski) [reply notebooks/MEMA_episode_visualisation.ipynb:10]: done

File: notebooks/Middle_ear_Inter_subject_analysis.ipynb (hunk 5 above)
Comment (lkorczowski) [reply notebooks/Middle_ear_Inter_subject_analysis.ipynb]: done

**Pull request merged**